lib/awful/util.lua

```diff
@@ -253,16 +253,7 @@ end
 -- @tparam string path
 -- @treturn bool True if path exists and is a directory.
 function util.is_dir(path)
-    local file = io.open(path)
-    if file then
-        if not file:read(0) -- Not a regular file (might include empty ones).
-            and file:seek("end") ~= 0 then  -- And not a file with size 0.
-            io.close(file)
-            return true
-        end
-        io.close(file)
-    end
-    return false
+    return Gio.File.new_for_path(path):query_file_type({}) == "DIRECTORY"
```

Comment:
should i free it before returning?
https://developer.gnome.org/gio/stable/GFile.html#g-file-new-for-path
> Returns
> a new GFile for the given path . Free the returned object with g_object_unref().